src/storage.c

```
@@ -4,6 +4,9 @@
 #include "os.h"
 
 void get_secret(uint8_t* secret) {
+    if (secret == NULL) {
```

Are failing silently here?